packages/components/src/DatePicker/DatePicker.tsx

```diff
@@ -116,11 +116,14 @@ export function DatePicker({
     [styles.inline]: inline,
   });
   const { pickerRef } = useEscapeKeyToCloseDatePicker(open, ref);
+  const shouldFocusOnSelectedDate = smartAutofocus && open;
 
-  if (smartAutofocus) {
-    useRefocusOnActivator(open);
-    useEffect(focusOnSelectedDate, [open]);
-  }
+  useRefocusOnActivator(shouldFocusOnSelectedDate);
```

Comment:
`shouldFocusOnSelectedDate` will always be false if `smartAutoFocus` is false. It will be the value of `open` is `smartAutoFocus` is true (just as before)